src/app/shared/components/validation-hint/validation-hint.component.ts

```diff
@@ -51,30 +51,45 @@ export class ValidationHintComponent implements OnInit, OnDestroy, OnChanges {
   constructor(private cdr: ChangeDetectorRef) {}
 
   public ngOnInit(): void {
-    this.validationFormControl.statusChanges.pipe(debounceTime(200), takeUntil(this.destroy$)).subscribe(() => {
-      const errors = this.validationFormControl.errors;
+    if (this.validationFormControl instanceof FormControl) {
```

Comment:
Can we rewrite this part in the way shown below?

```
this.validationFormControl.statusChanges.pipe(debounceTime(200), takeUntil(this.destroy$)).subscribe(() => {
  if (this.validationFormControl?.controls.length) {
    Object.keys(this.validationFormControl.controls).forEach(() => { ... })
  } else {
    this.updateValidationState(this.validationFormControl as FormControl);
  }
});
```

Reply:
We can't use this condition:
`if (this.validationFormControl?.controls.length)`
because FormGroup.controls doesn't have the `length` property.
However, putting the condtitional statement inside of `subscribe()` sounds great.